### PR TITLE
Normalizing Online-DDL queries

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Backport
NO

## Status
REAFY

## Description

In this PR we normalize some DDL queries sent as Online DDL:

- `CREATE INDEX` formats onto `ALTER TABLE`: this is because `gh-ost` and `pt-osc` only know how to handle `ALTER TABLE` statements. As there is no specific advantage to using `CREATE INDEX` over the equivalent `CREATE TABLE... ADD INDEX` we choose to have `CreateIndex.Format()` function export as `ALTER TABLE`
- `DROP TABLE <multi>` - exploded into multiple single-table `DROP` statements

## Related Issue(s)

https://github.com/vitessio/vitess/issues/6926#issuecomment-741711414

## Todos
- [x] Tests
- [x] Documentation

## Impacted Areas in Vitess
List general components of the application that this PR will affect:

- [x]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build 

## Notes

@GuptaManan100 @systay please notice the following test from `parse_test.go`:

```
		input: "create index a on b (col1) using btree key_block_size=12 with parser 'a' comment 'string' algorithm inplace lock none",
		// This is actually an incorrect output: should be the one commented
		output: "alter table b add index a (col1) using btree key_block_size 12 with parser 'a' comment 'string' algorithm inplace lock none",
		// output: "alter table b add index a using btree (col1) key_block_size=12 with parser 'a' comment 'string' algorithm inplace lock none",
```

The output is incorrect; still to understand why, I think the problem is with the parser, not with the Format. FWIW MySQL itself is unhappy about that complex CREAT EINDEX query.

### CODE REVIEW

File: go/vt/schema/parser_test.go [hunk 1]
```diff
@@ -40,14 +42,44 @@ func TestParseAlterTableOptions(t *testing.T) {
...
+		"create index i_idx on t(name(12))":             {sqls: []string{"alter table t add index i_idx (`name`(12))"}},
+		"create index i_idx on t(id, `ts`, name(12))":   {sqls: []string{"alter table t add index i_idx (id, ts, `name`(12))"}},
+		"create unique index i_idx on t(id)":            {sqls: []string{"alter table t add unique index i_idx (id)"}},
+		"create index i_idx using btree on t(id)":       {sqls: []string{"alter table t add index i_idx using btree (id)"}},
+		"create index with syntax error i_idx on t(id)": {isError: true},
+		"select * from t":                               {isError: true},
+		"drop database t":                               {isError: true},
+	}
+	for query, expect := range tests {
+		normalized, err := NormalizeOnlineDDL(query)
```
Comment (systay) [inline go/vt/schema/parser_test.go:73]: nit:
```golang
for query, expect := range tests {
	t.Run(query, func(t *testing.T) {
...
	}
}
})
```

sub tests makes it look good in test output

Comment (shlomi-noach) [reply go/vt/schema/parser_test.go:73]: TIL actually

**Comment by shlomi-noach** [comment]
Still have some tests to fix.

**Review by GuptaManan100 — APPROVED** [review]
LGTM 👍

File: go/vt/schema/parser_test.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/schema/parser_test.go]: fixed

**Comment by shlomi-noach** [comment]
Documentation: https://github.com/vitessio/website/pull/638

**Pull request merged**